src/2-run-ipfs-on-a-vps/index.md

```diff
@@ -0,0 +1,142 @@
+    peer identity: QmSyPpT59gXxtnLRZePQBthJd934iy17bmQesgHUAw25pB
+    to get started, enter:
+
+      ipfs cat /ipfs/QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG/readme
+    ```
+
+Note that your peer ID will be different, and that your node init file will be in your default user directory.
+
+IPFS works by actively seeking nearby nodes to connect to, which is a good thing for performance and availability, particularly in home and office networks. This causes addresses in the networks to be dialed that may not be there. Unfortunately, some VPS providers incorrectly classify this as suspicious activity, and some even have blocked nodes for doing so. To avoid this, let's add two things to the config file:
+
+    ```sh
+    # 1. disable mDNS discovery
+    ipfs config --json Discovery.MDNS.Enabled false
+
+    # 2. filter out local network addresses
+    ipfs config --json Swarm.AddrFilters '[
+      "/ip4/10.0.0.0/ipcidr/8",
+      "/ip4/100.64.0.0/ipcidr/10",
+      "/ip4/169.254.0.0/ipcidr/16",
+      "/ip4/172.16.0.0/ipcidr/12",
+      "/ip4/192.0.0.0/ipcidr/24",
+      "/ip4/192.0.0.0/ipcidr/29",
+      "/ip4/192.0.0.8/ipcidr/32",
+      "/ip4/192.0.0.170/ipcidr/32",
+      "/ip4/192.0.0.171/ipcidr/32",
+      "/ip4/192.0.2.0/ipcidr/24",
+      "/ip4/192.168.0.0/ipcidr/16",
+      "/ip4/198.18.0.0/ipcidr/15",
+      "/ip4/198.51.100.0/ipcidr/24",
+      "/ip4/203.0.113.0/ipcidr/24",
+      "/ip4/240.0.0.0/ipcidr/4"
+    ]'
+    ```
+
+Now you're ready to start IPFS!
+
+```sh
+> ipfs daemon &
+[1] 16252
+Initializing daemon...
+Adjusting current ulimit to 1024.
+> Swarm listening on /ip4/127.0.0.1/tcp/4001
+Swarm listening on /ip4/172.20.20.20/tcp/4001
+Swarm listening on /ip4/73.114.34.208/tcp/37131
+Swarm listening on /ip6/::1/tcp/4001
+API server listening on /ip4/127.0.0.1/tcp/5001
+Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
+Daemon is ready
+```
+
+This will run your daemon in the background, so you won't need to switch to a new window. You can make sure it is running using the `jobs` command.
+
+Give it a minute to connect to some other IPFS nodes, and then test that it's working by running a quick test:
+
+```sh
+> echo "hello world" | ipfs add
+added QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+```
+
+Now run this command to make sure that your IPFS node had this content locally:
+
+```sh
+> ipfs refs local | grep QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+```
+
+To check if your IPFS node has this content pinned, you can run `ipfs pin ls`:
+
+```
+> ipfs pin ls | grep QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o
+QmT78zSuBmuS4z925WZfrqQ1qHaJ56DQaTfyMUF7F8ff5o recursive
+```
+
+In this example, the hash is returned because `ipfs add` pins a file passed to it by default. If the hash is not returned, then the content will be removed at the next garbage collection. To stop this from happening, let's pin it:
```

Comment:
Should probably make it a bit more clear here that `ipfs add` does actually pin content automatically